src/main/java/uk/gov/hmcts/darts/util/DataUtil.java

```diff
@@ -15,13 +16,30 @@ public static String toUpperCase(String value) {
         return Optional.ofNullable(value).map(String::toUpperCase).orElse(null);
     }
 
+    public static List<String> trim(List<String> list) {
+        return Optional.ofNullable(list)
+            .map(l -> l.stream()
+                .map(DataUtil::trim)
+                .toList())
+            .orElse(null);
+    }
```

Comment:
Opinion, but I find this hard to grok. I think an explicit null check is more readable and easier to debug as it avoids the Optional wrapper and the nested stream:

```java
        if (list == null) {
            return null;
        }

        return list.stream()
            .map(DataUtil::trim)
            .toList();
```

Reply:
Updated :)